src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -816,29 +814,28 @@ public unsafe void CopyTo(Array array, int index)
                 Vector128<byte> lowerShuffleMask_CopyToBoolArray = Vector128.Create(0, 0x01010101_01010101).AsByte();
                 Vector128<byte> upperShuffleMask_CopyToBoolArray = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
 
-                if (Avx2.IsSupported)
+                if (Vector256.IsHardwareAccelerated)
                 {
                     Vector256<byte> shuffleMask = Vector256.Create(lowerShuffleMask_CopyToBoolArray, upperShuffleMask_CopyToBoolArray);
```

Comment:
This is going to rely on the JIT doing "several" optimizations to be efficient and "may" pessimize certain scenarios due to current JIT limitations.

We should really just have `Vector256<byte> shuffleMask = Vector256.Create(0x0000_0000_0000_0000, 0x0101_0101_0101_0101, 0x02020202_02020202, 0x03030303_03030303).AsByte();` and then below likewise just initialize `lowerShuffleMask` and `upperShuffleMask` directly.

Reply:
Actually noting that this should be `Vector256.Create(0x0000_0000_0000_0000, 0x0101_0101_0101_0101, 0x12121212_12121212, 0x13131313_13131313).AsByte();`.

As explained in https://github.com/dotnet/runtime/issues/72793#issuecomment-1194425845, `Vector256.Shuffle` treats it as `1x256-bit vector` not as `2x128-bit vector` like `Avx2` does and so you need to offset the "upper" elements by `Vector128<T>.Count` to ensure the "correct" indices are selected.

This is "logically the same" since all `uint` of the input vector are identical, but the JIT isn't advanced enough to know or track that.

Reply:
> Vector256.Shuffle treats it as 1x256-bit vector not as 2x128-bit vector like Avx2 does and so you need to offset the "upper" elements by Vector128<T>.Count to ensure the "correct" indices are selected.

@tannergooding  Thanks a lot! That was very helpful!